l1-contracts/test/portals/RNA.t.sol

```diff
@@ -0,0 +1,61 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright 2023 Aztec Labs.
+pragma solidity >=0.8.18;
+
+import {Test} from "forge-std/Test.sol";
+
+import {Rollup} from "@aztec/core/Rollup.sol";
+import {Outbox} from "@aztec/core/messagebridge/Outbox.sol";
+import {Inbox} from "@aztec/core/messagebridge/Inbox.sol";
+import {Registry} from "@aztec/core/messagebridge/Registry.sol";
+
+import {RollupNativeAsset} from "./RollupNativeAsset.sol";
+
+contract RNATest is Test {
+  Rollup internal rollup;
+  RollupNativeAsset internal rna;
+
+  Registry internal registry;
+  Outbox internal outbox;
+  Inbox internal inbox;
+
+  // Using a run of e2e_rollup_native_asset_contract.test.ts to get values
+  // for entryKeys and aztecAddress
+  bytes32 internal constant AZTEC_ADDRESS =
+    0x06b5fb872b0b08560791085e14039c2c23c3ba6591cc21ae067cd9036461d032;
+  bytes32 internal constant ENTRY_KEY =
+    0x1aa9c93cf0144c36f3242e9983e11eb153bbc401340c8777e49e28093ac88b86;
+
+  function setUp() public {
+    registry = new Registry();
+    outbox = new Outbox(address(registry));
+    inbox = new Inbox(address(registry));
+    rollup = new Rollup(registry);
+
+    registry.setAddresses(address(rollup), address(inbox), address(outbox));
+
+    rna = new RollupNativeAsset();
+    vm.etch(address(0xbeef), address(rna).code);
```

Comment:
guessing this is hardcoded for the entry key calculation?

Reply:
Ye, this was just to match what was generated from the typescript, etching to make it a bit easier to match.

Reply:
can you add a comment on why the `etch`?